---
title: "TAM_project_BMB502"
output: html_document
date: "2025-04-15"
---

#Load libraries

```{r}

library(DESeq2)
library(pheatmap)
library(EnhancedVolcano)
library(org.Mm.eg.db)
```

#Read in and reformat read counts

```{r}
# Read in the raw read counts
rawCounts <- read.delim("merged_readcounts.txt",header=TRUE,sep="\t")
head(rawCounts)

# Reformat the data so that the rownames are Ensembl Gene IDs
rnaseqMatrix <- rawCounts[,c(2:13)]
rownames(rnaseqMatrix) <- rawCounts[,1]
head(rnaseqMatrix)
```

#Create Metadata table

```{r}
# Define how the samples map to different sample groups
samples <- data.frame(matrix(c("con_AP1", "con_AP2", "con_AP3", "TAM_AP1", "TAM_AP2", "TAM_AP3","con_PP1", "con_PP2", "con_PP3", "TAM_PP1", "TAM_PP2", "TAM_PP3","con_AP", "con_AP", "con_AP", "TAM_AP", "TAM_AP", "TAM_AP","con_PP", "con_PP", "con_PP", "TAM_PP", "TAM_PP", "TAM_PP","con", "con", "con", "TAM", "TAM", "TAM","con", "con", "con", "TAM", "TAM", "TAM"), ncol=3))

names(samples) <- c("ID", "Tissue","Treatment")
rownames(samples) <- samples[,1]

samples$Tissue <- factor(c("con_AP", "con_AP", "con_AP", "TAM_AP", "TAM_AP", "TAM_AP","con_PP", "con_PP", "con_PP", "TAM_PP", "TAM_PP", "TAM_PP"))

samples$Treatment <- factor(c("con", "con", "con", "TAM", "TAM", "TAM","con", "con", "con", "TAM", "TAM", "TAM"))

```

#DESeq2- design by treatment

```{r}
# Create the DEseq2DataSet object
TAM_vs_con_deseq2Data <- DESeqDataSetFromMatrix(countData = rnaseqMatrix,
                              colData = samples,
                              design = ~ Treatment)
Tissue_deseq2Data <- DESeqDataSetFromMatrix(countData = rnaseqMatrix,
                              colData = samples,
                              design = ~ Tissue)

# Determine how many genes were lowly expressed and decide on a threshold
dim(TAM_vs_con_deseq2Data)
dim(TAM_vs_con_deseq2Data[rowSums(counts(TAM_vs_con_deseq2Data)) > 10, ])

dim(Tissue_deseq2Data)
dim(Tissue_deseq2Data[rowSums(counts(TAM_vs_con_deseq2Data)) > 10, ])

# Filter out lowly expressed genes
TAM_vs_con_deseq2Data <- TAM_vs_con_deseq2Data[rowSums(counts(TAM_vs_con_deseq2Data)) > 10, ]

Tissue_deseq2Data <- Tissue_deseq2Data[rowSums(counts(Tissue_deseq2Data)) > 10, ]
```

#All samples pairwise scatterplot

```{r}
# Pairwise scatterplots (THIS CAN TAKE A LONG TIME TO RUN)
pdf(file.path("All_Samples","TAM_con_scatterplots.pdf"))
pairs(log2(counts(TAM_vs_con_deseq2Data)+1), pch=16)
dev.off()
```

# differential expression, annotating and log transformation

```{r}
# Run pipeline for differential expression steps
TAM_vs_con_deseq2Data <- DESeq(TAM_vs_con_deseq2Data)
res_TAM_vs_con <- results(TAM_vs_con_deseq2Data)

Tissue_deseq2Data <- DESeq(Tissue_deseq2Data)
res_Tissue <- results(Tissue_deseq2Data)

#remove rows with NA pvalues
res_TAM_vs_con <- na.omit(res_TAM_vs_con)
res_Tissue <- na.omit(res_Tissue)

#sort by pvalue 
resOrdered_TAM_vs_con <- res_TAM_vs_con[order(res_TAM_vs_con$pvalue),]
df_TAM_vs_con <- data.frame(resOrdered_TAM_vs_con)
write.csv(df_TAM_vs_con, file = file.path("All_Samples", "DESeq_results_TAM_vs_con.csv"))


resOrdered_Tissue <- res_Tissue[order(res_Tissue$pvalue),]
df_Tissue <- data.frame(resOrdered_Tissue)


# rlog transform counts
Treatment_rld <- rlog(TAM_vs_con_deseq2Data, blind=FALSE)
Treatment_rlogcounts <- data.frame(assay(TAM_vs_con_deseq2Data))
rownames(Treatment_rlogcounts) <- rownames(TAM_vs_con_deseq2Data)

Tissue_rld <- rlog(Tissue_deseq2Data, blind=FALSE)
Tissue_rlogcounts <- data.frame(assay(Tissue_deseq2Data))
rownames(Tissue_rlogcounts) <- rownames(Tissue_deseq2Data)

#Annotate Traetment DESeq2 with gene symbols
symbols_Treatment_deseq2Data<- TAM_vs_con_deseq2Data

library(org.Mm.eg.db)
ens <- rownames(symbols_Treatment_deseq2Data)
symbols <- mapIds(org.Mm.eg.db, keys=ens, column='SYMBOL',
                  keytype='ENSEMBL')
symbols <- symbols[match(rownames(symbols_Treatment_deseq2Data), names(symbols))]
rownames(symbols_Treatment_deseq2Data) <- symbols
symbols_Treatment_deseq2Data <- symbols_Treatment_deseq2Data[!is.na(rownames(symbols_Treatment_deseq2Data)),]
head(symbols_Treatment_deseq2Data)
res_symbols_Treatment_deseq2Data <- results(symbols_Treatment_deseq2Data, contrast = c("Treatment","TAM","con"))

#remove rows with NA pvalues
res_symbols_Treatment_deseq2Data <- na.omit(res_symbols_Treatment_deseq2Data)
df_res_symbols_Treatment_deseq2Data <- data.frame(res_symbols_Treatment_deseq2Data)


```

#All samples PCA plot

```{r}
# PCA plot data
PCA_data<-plotPCA(Tissue_rld, intgroup=c("Tissue"), returnData=TRUE)

# PCA plot
plotPCA(Tissue_rld, intgroup=c("Tissue"))

#Write plot into PDF
pdf(file.path("All_Samples","PCA_plot_TAM_vs_con.pdf"))
plotPCA(Tissue_rld, intgroup=c("Tissue"))

dev.off()
```

#Plot and heatmap for log tranformed data

```{r}
# Inspect rlog data using scatterplots (THIS CAN TAKE A LONG TIME TO RUN)
pdf(file.path("All_Samples","TAM_vs_con_scatterplots_rlog.pdf"))
pairs(log2(Treatment_rlogcounts+1), pch=16)
dev.off()

# Inspect rlog data using heatmap of pairwise correlation
# (THIS CAN TAKE A LONG TIME TO RUN)
pdf(file.path("All_Samples","TAM_vs_con_heatmap_rlog_correlation.pdf"))
pheatmap(cor(assay(Treatment_rld)))
dev.off()
```

# Signifigance Data for all samples

```{r}

# Get number of differentially expressed data at different thresholds
summary(res_TAM_vs_con)
sum(res_TAM_vs_con$padj < 0.05, na.rm=TRUE)
sum(res_TAM_vs_con$pvalue < 0.05, na.rm=TRUE)
```

```{r}
# Identify top 3 genes with largest fold change, list gene symbols for probe set ids
top_genes <- head(df_res_symbols_Treatment_deseq2Data[order(df_res_symbols_Treatment_deseq2Data$log2FoldChange, decreasing = TRUE),], 3)
top_genes_symbols <- rownames(top_genes)
  
# Identify top 3 genes with smallest fold change, list gene symbols for probe set ids
bottom_genes <- tail(df_res_symbols_Treatment_deseq2Data[order(df_res_symbols_Treatment_deseq2Data$log2FoldChange, decreasing = TRUE),], 3)
bottom_genes_symbols <- rownames(bottom_genes)

# MA plot
pdf(file.path("All_Samples","TAM_vs_con__MA_plot.pdf"))
plotMA(res_symbols_Treatment_deseq2Data, ylim= c(-5, 5))
# Label top genes
text(x = top_genes$baseMean, y = top_genes$log2FoldChange, 
     labels = rownames(top_genes), pos = 1, cex = 0.6, col = "red")
# Label bottom genes
text(x = bottom_genes$baseMean, y = bottom_genes$log2FoldChange,
     labels = rownames(bottom_genes), pos = 4, cex = 0.6, col = "green")
dev.off()

# Volcano plot
res_symbols_Treatment_deseq2Data <- results(symbols_Treatment_deseq2Data, contrast = c("Treatment","TAM","con"))
res_TAM_vs_con_lfc<- lfcShrink(symbols_Treatment_deseq2Data, contrast = c("Treatment","TAM","con"), res=res_symbols_Treatment_deseq2Data, type='normal')

pdf(file.path("All_Samples","TAM_vs_con_volcano_plot.pdf"))
EnhancedVolcano(res_TAM_vs_con_lfc,
                lab = rownames(res_TAM_vs_con_lfc),
                x = 'log2FoldChange',
                y = 'pvalue',
                pCutoff = 1e-9,
                drawConnectors = TRUE,
                widthConnectors = 0.5,
                FCcutoff = 1.0,
                pointSize = 2.0,
                labSize = 3.0)
dev.off() 
```


```{r}
#Annotate Tissue DESeq2 with gene symbols
symbols_Tissue_deseq2Data<- Tissue_deseq2Data

library(org.Mm.eg.db)
ens <- rownames(symbols_Tissue_deseq2Data)
symbols <- mapIds(org.Mm.eg.db, keys=ens, column='SYMBOL',
                  keytype='ENSEMBL')
symbols <- symbols[match(rownames(symbols_Tissue_deseq2Data), names(symbols))]
rownames(symbols_Tissue_deseq2Data) <- symbols
symbols_Tissue_deseq2Data <- symbols_Tissue_deseq2Data[!is.na(rownames(symbols_Tissue_deseq2Data)),]
head(symbols_Tissue_deseq2Data)

rld_symbols_Tissue_deseq2Data <- rlog(symbols_Tissue_deseq2Data, blind=FALSE)
rlogcounts_symbols_Tissue_deseq2Data <- data.frame(assay(symbols_Tissue_deseq2Data))
#rownames(rlogcounts_symbols_Tissue_deseq2Data) <- rownames(symbols_Tissue_deseq2Data)
```

```{r}
# Comparing TAM_AP vs con_AP
res_TAM_AP_vs_con_AP <- results(symbols_Tissue_deseq2Data, contrast=c("Tissue", "TAM_AP", "con_AP"))

res_TAM_AP_vs_con_AP <- lfcShrink(symbols_Tissue_deseq2Data, contrast=c("Tissue", "TAM_AP", "con_AP"), res=res_TAM_AP_vs_con_AP, type = 'normal')

#remove rows with NA pvalues
res_TAM_AP_vs_con_AP <- na.omit(res_TAM_AP_vs_con_AP)
df_res_TAM_AP_vs_con_AP <- data.frame(res_TAM_AP_vs_con_AP)

resOrdered_TAM_AP_vs_con_AP <- res_TAM_AP_vs_con_AP[order(res_TAM_AP_vs_con_AP$pvalue),]
write.csv(df_res_TAM_AP_vs_con_AP, file =file.path("TAM_AP_vs_con_AP","DESeq_results_TAM_AP_vs_con_AP.csv"))

summary(res_TAM_AP_vs_con_AP)
  # Get number of differentially expressed data at different thresholds
sum(res_TAM_AP_vs_con_AP$padj < 0.05, na.rm=TRUE)
sum(res_TAM_AP_vs_con_AP$pvalue < 0.05, na.rm=TRUE)

# Identify top 3 genes with largest fold change, list gene symbols for probe set ids
top_genes <- head(df_res_TAM_AP_vs_con_AP[order(df_res_TAM_AP_vs_con_AP$log2FoldChange, decreasing = TRUE),], 3)
top_genes_symbols <- rownames(top_genes)
  
# Identify top 3 genes with smallest fold change, list gene symbols for probe set ids
bottom_genes <- tail(df_res_TAM_AP_vs_con_AP[order(df_res_TAM_AP_vs_con_AP$log2FoldChange, decreasing = TRUE),], 3)
bottom_genes_symbols <- rownames(bottom_genes)

# MA plot
pdf(file.path("TAM_AP_vs_con_AP","TAM_AP_vs_con_AP_MA_plot.pdf"))
plotMA(res_TAM_AP_vs_con_AP, ylim= c(-2.5, 2.5))
# Label top genes
text(x = top_genes$baseMean, y = top_genes$log2FoldChange, 
     labels = rownames(top_genes), pos = 4, cex = 0.6, col = "red")
# Label bottom genes
text(x = bottom_genes$baseMean, y = bottom_genes$log2FoldChange,
     labels = rownames(bottom_genes), pos = 4, cex = 0.6, col = "green")
dev.off()



# Volcano plot
pdf(file.path("TAM_AP_vs_con_AP","TAM_AP_vs_con_AP_Volcano_plot.pdf"))
EnhancedVolcano(res_TAM_AP_vs_con_AP,
lab = rownames(res_TAM_AP_vs_con_AP),
x = 'log2FoldChange',
y = 'pvalue',
pCutoff = 0.000000001,
drawConnectors = TRUE,
widthConnectors = 0.5,
FCcutoff = 1.0,
pointSize = 2.0,
labSize = 3.0)
dev.off()

#Heat map
# Candidate genes
candidates <- read.table("AP_Top_Up_Down.txt",header=FALSE)
names(candidates) <- c("Symbol")

# Subset results with data just for candidate genes
table_hits <- rlogcounts_symbols_Tissue_deseq2Data[candidates$Symbol,]

# Replace any NA's with 0
table_hits[is.na(table_hits)] <- 0

# Replace Ensembl Gene IDs with symbols
#row.names(table_hits) <- candidates$Symbol

cal_z_score <- function(x){
  (x - mean(x)) / sd(x)
}

table_hits_norm <- t(apply(table_hits, 1, cal_z_score))

pheatmap(table_hits_norm,name = "Row Z-Score",)

pdf(file.path("TAM_AP_vs_con_AP","TAM_AP_vs_con_AP_heat_map_plot.pdf"))
pheatmap(table_hits_norm,name = "Row Z-Score",)
dev.off()
```

```{r}
# Comparing TAM_PP vs con_PP
res_TAM_PP_vs_con_PP <- results(symbols_Tissue_deseq2Data, contrast=c("Tissue", "TAM_PP", "con_PP"))

res_TAM_PP_vs_con_PP <- lfcShrink(symbols_Tissue_deseq2Data, contrast=c("Tissue", "TAM_PP", "con_PP"), res=res_TAM_PP_vs_con_PP, type = 'normal')

#remove rows with NA pvalues
res_TAM_PP_vs_con_PP <- na.omit(res_TAM_PP_vs_con_PP)
df_res_TAM_PP_vs_con_PP <- data.frame(res_TAM_PP_vs_con_PP)

resOrdered_TAM_PP_vs_con_PP <- res_TAM_PP_vs_con_PP[order(res_TAM_PP_vs_con_PP$pvalue),]
write.csv(df_res_TAM_PP_vs_con_PP, file =file.path("TAM_PP_vs_con_PP","DESeq_results_TAM_PP_vs_con_PP.csv"))

summary(res_TAM_PP_vs_con_PP)
  # Get number of differentially expressed data at different thresholds
sum(res_TAM_PP_vs_con_PP$padj < 0.05, na.rm=TRUE)
sum(res_TAM_PP_vs_con_PP$pvalue < 0.05, na.rm=TRUE)

  # MA plot
pdf(file.path("TAM_PP_vs_con_PP","TAM_PP_vs_con_PP_MA_plot.pdf"))
plotMA(res_TAM_PP_vs_con_PP)
dev.off()

# Volcano plot
pdf(file.path("TAM_PP_vs_con_PP","TAM_PP_vs_con_PP_Volcano_plot.pdf"))
EnhancedVolcano(res_TAM_PP_vs_con_PP,
lab = rownames(res_TAM_PP_vs_con_PP),
x = 'log2FoldChange',
y = 'pvalue',
pCutoff = 0.000000001,
drawConnectors = TRUE,
widthConnectors = 0.5,
FCcutoff = 1.0,
pointSize = 2.0,
labSize = 3.0)
dev.off()

#Heat map
# Candidate genes
candidates <- read.table("PP_Top_Up_Down.txt",header=FALSE)
names(candidates) <- c("Symbol")

# Subset results with data just for candidate genes
table_hits <- rlogcounts_symbols_Tissue_deseq2Data[candidates$Symbol,]

# Replace any NA's with 0
table_hits[is.na(table_hits)] <- 0

# Replace Ensembl Gene IDs with symbols
#row.names(table_hits) <- candidates$Symbol

cal_z_score <- function(x){
  (x - mean(x)) / sd(x)
}

table_hits_norm <- t(apply(table_hits, 1, cal_z_score))

pheatmap(table_hits_norm,name = "Row Z-Score",)

pdf(file.path("TAM_PP_vs_con_PP","TAM_AP_vs_TAM_PP_heat_map_plot.pdf"))
pheatmap(table_hits_norm,name = "Row Z-Score",)
dev.off()

```


```{r}
# Comparing con_AP vs con_PP
res_con_PP_vs_con_AP <- results(symbols_Tissue_deseq2Data, contrast=c("Tissue", "con_PP", "con_AP"))

#remove rows with NA pvalues
res_con_PP_vs_con_AP <- na.omit(res_con_PP_vs_con_AP)
df_res_con_PP_vs_con_AP <- data.frame(res_con_PP_vs_con_AP)

resOrdered_con_PP_vs_con_AP <- res_con_PP_vs_con_AP[order(res_con_PP_vs_con_AP$pvalue),]
write.csv(df_res_con_PP_vs_con_AP, file =file.path("con_PP_vs_con_AP","DESeq_results_con_PP_vs_con_AP.csv"))

summary(res_con_PP_vs_con_AP)
  # Get number of differentially expressed data at different thresholds
sum(res_con_PP_vs_con_AP$padj < 0.05, na.rm=TRUE)
sum(res_con_PP_vs_con_AP$pvalue < 0.05, na.rm=TRUE)

  # MA plot
pdf(file.path("con_PP_vs_con_AP","con_PP_vs_con_AP_MA_plot.pdf"))
plotMA(res_con_PP_vs_con_AP)
dev.off()

# Volcano plot
pdf(file.path("con_PP_vs_con_AP","con_PP_vs_con_AP_Volcano_plot.pdf"))
EnhancedVolcano(res,
lab = rownames(res),
x = 'log2FoldChange',
y = 'pvalue',
pCutoff = 0.000000001,
drawConnectors = TRUE,
widthConnectors = 0.5,
FCcutoff = 1.0,
pointSize = 2.0,
labSize = 3.0)
dev.off()

#Heat map
pdf(file.path("con_PP_vs_con_AP","con_PP_vs_con_AP_heat_map_plot.pdf"))
pheatmap(res_con_PP_vs_con_AP,name = "Row Z-Score",)
dev.off()
```

```{r}
# Comparing TAM_AP vs TAM_PP
res_TAM_AP_vs_TAM_PP <- results(symbols_Tissue_deseq2Data, contrast=c("Tissue", "TAM_AP", "TAM_PP"))

#remove rows with NA pvalues
res_TAM_AP_vs_TAM_PP <- na.omit(res_TAM_AP_vs_TAM_PP)
df_res_TAM_AP_vs_TAM_PP <- data.frame(res_TAM_AP_vs_TAM_PP)

resOrdered_TAM_AP_vs_TAM_PP <- res_TAM_AP_vs_TAM_PP[order(res_TAM_AP_vs_TAM_PP$pvalue),]
write.csv(df_res_TAM_AP_vs_TAM_PP, file =file.path("TAM_PP_vs_TAM_AP","DESeq_results_TAM_AP_vs_TAM_PP.csv"))

summary(res_TAM_AP_vs_TAM_PP)
  # Get number of differentially expressed data at different thresholds
sum(res_TAM_AP_vs_TAM_PP$padj < 0.05, na.rm=TRUE)
sum(res_TAM_AP_vs_TAM_PP$pvalue < 0.05, na.rm=TRUE)

  # MA plot
pdf(file.path("TAM_PP_vs_TAM_AP","TAM_AP_vs_TAM_PP_MA_plot.pdf"))
plotMA(res_TAM_AP_vs_TAM_PP)
dev.off()

# Volcano plot
pdf(file.path("TAM_PP_vs_TAM_AP","TAM_AP_vs_TAM_PP_Volcano_plot.pdf"))
EnhancedVolcano(res,
lab = rownames(res),
x = 'log2FoldChange',
y = 'pvalue',
pCutoff = 0.000000001,
drawConnectors = TRUE,
widthConnectors = 0.5,
FCcutoff = 1.0,
pointSize = 2.0,
labSize = 3.0)
dev.off()

#Heat map
pdf(file.path("TAM_PP_vs_TAM_AP","TAM_AP_vs_TAM_PP_heat_map_plot.pdf"))
pheatmap(res_TAM_AP_vs_TAM_PP,name = "Row Z-Score",)
dev.off()
```